src/index.ts

```diff
@@ -13,14 +13,19 @@ const trimOsText = (version: string) => {
 };
 
 const safeSemver = (version: string) => {
-	return version.replace(/(\.[0-9]+)\.rev/, '$1+rev');
+	// fix major.minor.patch.rev to use rev as build metadata
+	return version.replace(/(\.[0-9]+)\.rev/, '$1+rev')
+		// fix major.minor.patch.prod to be treat .dev & .prod as build metadata
+		.replace(/([0-9]+\.[0-9]+\.[0-9]+)\.(dev|prod)((?:[-\.][0-9a-z]+)*)/i, '$1+$2$3')
```

Comment:
Nit, but you could remove everything after `(dev|prod)` here I think, because you don't need to match the whole string, just the part you want to replace. Right now you allow `1.2.3.dev.abc` but not `1.2.3.dev+abc`. Both are a bit weird, but personally I'd just ignore the difference and make this regex simpler. You could trim this beast down to

 ```regex
.replace(/([0-9]+\.[0-9]+\.[0-9]+)\.(dev|prod)\b/, '$1+$2')
```

That'll fix anything that has `.dev` or `.prod` followed by a word boundary (so punctuation, whitespace and the end of the string is fine, but `.devs` doesn't match).

Reply:
I actually had in mind to avoid converting `1.2.3.dev+abc` to `1.2.3+dev+abc`, since it's invalid.
On the other hand `1.2.3.dev+abc` is already invalid.

Reply:
How about changing `1.2.3.dev+abc` to `1.2.3+dev.abc`? I guess that it might be an overkill and possibly such a release simply doesn't exist.

Reply:
I don't think any versions like that exist, and I think this way is simpler. Up to you, but personally I'd just ignore it :smiley: